### DISCUSSION TITLE
👨‍💻☎ WinUI Community Call: September 21, 2022

### DISCUSSION THREAD

**Discussion by gabbybilka** [microsoft/microsoft-ui-xaml#7727] [Announcements]
### https://www.youtube.com/watch?v=iikSa_8c7Jk

# Details
Date: September 21, 2022
Time: 16:00-17:00 UTC (9:00-10:00am Pacific)

Anyone and everyone is welcome - no pre-registration is required.
This will be an informal interactive live stream directly with members of our engineering team.

# Format
The community call is a call among the WinUI team that is live-streamed onto YouTube. We present on new updates, share information, welcome guests, and answer your questions. In the September call we will be sharing info about the upcoming releases for WinAppSDK & WinUI, joined by @DmitriyKomin to chat about Media controls in WinUI 3, and give a high-level overview of the [Template Studio](https://github.com/microsoft/TemplateStudio) Visual Studio extension.

### **Agenda**
- Intro
- Feature spotlight: Media controls in WinUI 3 - @DmitriyKomin
- Feature spotlight: [Template Studio](https://github.com/microsoft/TemplateStudio)
- Release & roadmap updates for WinUI & Windows App SDK 
- Q&A

### **Q&A Code of Conduct**
- To help ensure Q&A remains beneficial for the majority of the call's developer audience, please consider whether your question could be framed so that it might be more broadly relevant to our larger developer audience by ensuring it is reasonably concise, broadly beneficial, well-suited for the fast-paced live Q&A format, or able to spark open ended discussion.  
- Please understand some questions are not well-suited for the fast-paced live Q&A format and should be filed as standalone [Q&A](https://github.com/microsoft/microsoft-ui-xaml/discussions/new?category=q-a&title=%5BYour+Question+Here%5D) discussions where they can be most appropriately handled. Excluding presentation topic requests, this generally includes requests for dedicated follow-ups or technically complex questions that require investigative time or expertise not included in this month's guest list.

### Leave us your questions in the comments on this issue or live in the YouTube chat during the stream!

**Comment by jschwizer99** [comment]
Can you report the status of the drag&drop issue fixes? Currently, I'm still using WinUI App SDK 1.0.3. All App SDK 1.1.x versions just crash too often. Some first fixes were scheduled for 1.2.x but I couldn't find any hints on the release notes of 1.2.0-preview1 or the related tickets.
Related tickets are https://github.com/microsoft/microsoft-ui-xaml/issues/7002, https://github.com/microsoft/microsoft-ui-xaml/issues/7231, https://github.com/microsoft/microsoft-ui-xaml/issues/7192.
When and in what version can we expect remedies for preview and release channels?

**Reply by JPTGamesAndApps** [reply]
Visual Studio 2022 does not produce .msixbundle files anymore (just multiple .msix) for WinAppSDK projects. It used to produce .msixbundle files for UWP projects (when bundling is checked). Is that a known limitation or something is wrong with the installation/configuration? The current workaround is to manually pack the .msix files. MS Store does not allow .msix if .msixbundle was previously uploaded.

**Reply by koenvd** [reply]
Issue is being tracked here https://github.com/microsoft/WindowsAppSDK/issues/1808 @JPTGamesAndApps . I had to move back to a setup with 2 projects where one is the packaging project. 

But would good to have an update about this one as well indeed.

**Reply by DarranRowe** [reply]
I guess that this is one of the reasons why the single project packaging extension is still classed as preview.
But I also had to change to the separate packaging project to get the bundles.

**Comment by nikolayvpavlov** [comment]
Can you comment why Windows App Sdk and WinUI 3 are never mentioned in the posts about new releases on Dot Net Blog?  At the same time, MAUI is getting lots of publicity and is in the spotlight for NET 7.

Are there plans to introduce a layer on top of Win32 for basic features like message handing and GDI operations?  Think of WinForm's Graphics and all related classes.  We should not have to call SetWindowSubclass for handling basic messages like WM_COMMAND, nor need to work with device contexts.  Support for other features like ImageLists should be a must as well.

**Reply by kmgallahan** [reply]
WinUI = Windows and doesn't involve .NET in any way when writing C++ apps, and only insomuch as C# relies on it when using that lang

In contrast WinForms, WPF, and MAUI are all dependent on .NET

**Reply by nikolayvpavlov** [reply]
@kmgallahan , WinForms is just a set of .NET classes on top of Win32.  It cannot exist by itself.  Like MFC, only done right.  If WinUI is not a first-class citizen in the .NET ecosystem, then it is as good as dead in the water.  C++ was never famous for being a productive tool for desktop apps.  If it were, VB, and C# w/ WinForms would never need to exist, let alone WPF and UWP.  

What is more, if WinUI doesn't become a first-class citizen in .NET, then Windows desktop development on .NET is doomed to obsoletion.

**Reply by kmgallahan** [reply]
MAUI is how you use .NET to target iOS, Android, macOS, and Windows (via WinUI). None of these native GUI technologies are talked about in .NET posts, because why would they be? 

The .NET team is responsible for making MAUI, thus they make announcements about it. They don't work on any of the native platform stacks, thus they have no reason to talk about WinUI anymore than iOS in their posts.

WinUI / WASDK / Win32 / DirectX / et.al. exist independent from .NET. Talking about them in .NET posts just because you can hit the API's from C# doesn't really promote them to being anything other than what they are.

**Reply by nikolayvpavlov** [reply]
You miss the point, don't you?  

Right now, what is the message to all .NET developers who do desktop apps?  "Stay with WPF and WinForms, because we are killing UWP, and forget about modern UI".  Change my mind.

I can't care less that Microsoft are busy with yet another API (after Win32, Shell COM, WinRT).  What I care is what should I use to modernize my NET 4.8 app when I migrate it to NET 7.

**Reply by kmgallahan** [reply]
> Right now, what is the message to all .NET developers who do desktop apps?
> What I care is what should I use to modernize my NET 4.8 app when I migrate it to NET 7.

If you are a ".NET developer" and only get your Windows development news from .NET blog posts then I guess you'll only be aware of:

- MAUI (which targets WinUI)
- Ability to switch to .NET 6/7 instead of .NET Framework
- Maybe they'll mention XAML islands when they work well

I however don't know any Windows developers that only read .NET blog posts.

Windows development guidance is clear: https://learn.microsoft.com/en-us/windows/apps/

**Reply by nikolayvpavlov** [reply]
Well thanks for proving my point - the link you give is focused on Windows App Sdk, and the tutorial - on C# and C++.  Now, tell me how .NET is not related.  

The fact is that Windows App Sdk gets close to zero publicity.

**Reply by DarranRowe** [reply]
To put it simply, .NET is peripherally related in that it is a client for the Windows App SDK.
The Windows App SDK itself is native C++ with the exception of Microsoft.WindowsAppRuntime.Release.Net.dll, which I think is the managed wrapper. This allows you to use the SDK without using Cs/WinRT.

Also, the Windows App SDK gets close to zero publicity? It is plastered all over the official Microsoft Windows 11 development documentation. If you are a new developer and you are looking at initial options, the documentation pushes you towards WinUI 3. To be frank, it is everywhere that matters, but also in the form that matters. It draws your attention to it allowing you to choose it, but it doesn't imply that it is your only choice.

**Comment by AathifMahir** [comment]
Any News on Native Splash Screen Support for WinUI 3 Apps Like UWP?

Currently different WinUI 3 Apps Goes with Custom Solution for Splash Screen that Makes WinUI 3 Apps Not Consistent and Bad At Initial User Experience

**Comment by ghost1372** [comment]
WinUI 3 App with an **InfoBar** and Publish as Single File Option will be Crash.
Why don't you fix this bug?
https://github.com/microsoft/microsoft-ui-xaml/issues/7143

**Comment by pubtom** [comment]
1.) When will you fix the unable to Interact with WebView in WinUI version > 2.5 issue?
WinUI 2.8 still has this issue! This is a blocking issue: WinUI2 is unusable since the 2.6 version if your app uses a WebView control.
I forced to use the WinUI 2.5 version because of this issue. Highly waiting for the fix!
see https://github.com/microsoft/microsoft-ui-xaml/issues/5299

2.) When will the WinUI 2.8 version of WebView2 run on UWP apps on Xbox?
see https://github.com/MicrosoftEdge/WebView2Feedback/issues/2155

**Comment by ghost1372** [comment]
Please provide a way to add a Context Menu item to the Windows 11 Context Menu, The current method (sparse package) is a bit complicated and difficult for us (C# developers who have not interacted with C++) and this method is not working with UnPackaged App.
Creating a special API for it will be very useful

https://github.com/microsoft/WindowsAppSDK/issues/1082

**Comment by ghost1372** [comment]
is there any news about new API for ApplicationData for Packaged/Unpackaged App?
https://github.com/microsoft/WindowsAppSDK/issues/2639

**Reply by DrusTheAxe** [reply]
No news. It's still in the backlog of ask/proposals for vFuture consideration. If this interests you please share your feedback on the Issue. Thanks

**Comment by koenvd** [comment]
- Do you have an idea already when drag and drop inside a Webview2 will be fixed (https://github.com/MicrosoftEdge/WebView2Feedback/issues/2805)?
- Any update available for the taskbar still visible when going fullscreen and the app is already maximised (https://github.com/microsoft/WindowsAppSDK/issues/2577)? Since comments mentioned the fix was planned to go out in one of the next servicing releases.

Thanks already!

**Comment by wbokkers** [comment]
When will the **drag&drop fixes** (available only in the latest preview) be available in a production release? We really need d&d.
Why are these fixes not available in the latest service release? What is the purpose of fixing bugs if we can't use it in production?

**Reply by wbokkers** [reply]
Thanks @bpulliam and @Scottj1s for answering these d&d related questions! Much appreciated.

**Comment by Skittles2519** [comment]
Any updates on WinUi2 WebView printing release window? Thank you

**Comment by ghost1372** [comment]
why productboard is not updated for v1.2?
 https://portal.productboard.com/winappsdk/1-windows-app-sdk/tabs/2-planned

**Comment by christosk92** [comment]
This has been an intrinsic problem to Windows apps since Windows 10 (or even 8), but virtualization (and thus scrolling) is just really slow when you have complex layouts. Now, obviously the complex layout is what's causing the issue. But other native frameworks (unrelated to .NET such as Swift/Flutter) have no trouble whatsoever rendering a very similar layout. 

This also ties in with other rendering performance (low fps) issues and many other memory related issues. Are there any plans for major performance improvements going forward?
